source/postcard-c/postcard.h

```diff
@@ -0,0 +1,1184 @@
+#ifndef POSTCARD_H
+#define POSTCARD_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+/// A growable view into a buffer
+typedef struct {
+    // Pointer to the underlying data buffer
+    uint8_t* data;
+    /// The total written length of data in the buffer
+    size_t len; // Current length of data
+    /// The maximum capacity of the buffer
+    size_t capacity;
+} postcard_slice_t;
+
+/// postcard_c error codes
+typedef enum {
```

Comment:
i know that this is currently pure C, but doesn't pulling it into a C++ code base pollute the namespace (unless you `#include` it within a `namespace`)?
would it make sense to make the header C & C++ and use `#ifdef __cplusplus` copiously to decide between the two? that'd e.g. allow using `enum class` rather than `typedef enum`